s3-handler/main.go

```diff
@@ -244,6 +265,15 @@ func main() {
 	if os.Getenv("FILTER_PATTERNS") != "" {
 		filterPatterns = strings.Split(os.Getenv("FILTER_PATTERNS"), ",")
 	}
+
+	if redactPatternString = os.Getenv("REDACT_PATTERN"); redactPatternString != "" {
+		redactPattern, err = regexp.Compile(redactPatternString)
+		if err != nil {
+			logrus.WithError(err).WithField("redact_pattern", redactPatternString).
```

Comment:
I think we should terminate if `REDACT_PATTERN` is set and the regex isn't compilable.  The implication being that the user intends to redact data, but messed up their env var, and if we keep processing we'll end up not redacting data as they desire.